main.py

```diff
@@ -84,6 +93,69 @@ def draw_overmap(screen, position):
         pg.draw.rect(screen, OVERMAP_BG, mask)
 
 
+# génération fruit
+pos_fruits = []
+col_fruits = []
+
+
+def inside_map(pos_fruit):
+    """test si le fruit généré est dans la map"""
+    return (
+        pos_fruit.x - WIDTH / 2 > 0
+        and pos_fruit.x - WIDTH / 2 < M_WIDTH
+        and pos_fruit.y - HEIGHT / 2 > 0
+        and pos_fruit.y - HEIGHT / 2 < M_HEIGHT
+    )
+
+
+def generate_random_fruit_position(position):
+    """génère une position aléatoire"""
+    while True:
+        px = int(position.x)
+        py = int(position.y)
+        x, y = random.randint(px, px + WIDTH), random.randint(py, py + HEIGHT)
+        pos_fruit = V2(x, y)
+        if inside_map(pos_fruit):
+            break
+
+    pos_fruits.append(pos_fruit)
+
+
+def generate_fruit(pos_fruits, position):
+    """génère un fruit aléatoirement sur le screen"""
+    if pos_fruits == [] or (random.random() > 0.99 and len(pos_fruits) < 20):
+        generate_random_fruit_position(position)
+        color = generate_random_color()
+        col_fruits.append(color)
+
+
+# affichage fruits
+
+
+def draw_fruits(screen, position, pos_fruits, col_fruits):
+    """affiche les fruits"""
+    for pos_fruit, color in zip(pos_fruits, col_fruits):
+        centerx = pos_fruit.x - position.x
+        centery = pos_fruit.y - position.y
+        center = (centerx, centery)
+
+        pg.draw.circle(screen, color, center, 10)
+
+
+# Manger fruit
+
+
+def eat_fruit(position, pos_fruits, size) -> int:
+    """si le fruit est assez proche, le mange. Renvoie la nouvelle taille après absorbation d'un ou plusieurs fruits"""
+    for pos_fruit in pos_fruits:
+        if (position + SCREEN // 2 - pos_fruit).length() < size:
```

Comment:
Bien !

Il y a cependant un petit problème (il me semble), `position` correspond à la vrai position de l'objet (et non `position + SCREEN // 2`).

Du coup, je recommande cela : 
```suggestion
        if (position - pos_fruit).length() < size:
```

Reply:
https://github.com/mines-ap2022-groupe-3/agar.io/pull/22#discussion_r1022927007